src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DefaultObjectValidator.cs

```diff
@@ -84,7 +84,10 @@ public void Validate([NotNull] ModelValidationContext modelValidationContext)
                 // the same for all the elements of the array, we do not do GetValidators for each element,
                 // instead we just pass them over. See ValidateElements function.
                 var validatorProvider = validationContext.ModelValidationContext.ValidatorProvider;
-                validators = validatorProvider.GetValidators(modelExplorer.Metadata);
+                var validatorProviderContext = new ModelValidatorProviderContext(modelExplorer.Metadata);
```

Comment:
No, the pattern is slightly more verbose, but it should stay as-is. This pattern allows us to build in more kinds of optimizations in the future if we need to (reusing the context for one).